ARCs/ARC-1.md

```diff
@@ -0,0 +1,116 @@
+# ARC-1: ITS Hub Multi-Chain Token Linking
+
+## Metadata
+
+- **ARC ID**: 1
+- **Author(s)**: Milap Sheth
+- **Status**: Draft
+- **Created**: 2024-11-14
+- **Last Updated**: 2024-11-14
+- **Target Implementation**: Q4 2024
+
+## Summary
+
+Extend the Interchain Token Service (ITS) Hub to support linking existing tokens across different chains, potentially using different decimals. This improvement enables token transfers for custom token links across both EVM and non-EVM chains by supporting appropriate decimal scaling.
+
+## Motivation
+
+### Background
+
+The current ITS `DeployTokenManager` message type only supports linking existing tokens with identical decimals. This limitation is particularly problematic when dealing with non-EVM chains, which often cannot support the high decimal values (e.g., 18) common in EVM chains due to token amount type constraints.
+
+### Goals
+
+- Enable linking of existing tokens across connected chains
+- Support linking tokens with different decimal precisions
+
+## Detailed Design
+
+### Overview
+
+Two potential solutions are proposed to implement this enhancement:
+
+1. Modify the existing `DeployTokenManager` message type
+2. Introduce new message types for token registration and linking
+
+### Technical Specification
+
+#### Solution 1: Enhanced DeployTokenManager
+
+```solidity
+struct DeployTokenManager {
+    bytes32 tokenId;
+    uint256 tokenManagerType;
+    bytes params;
+    uint8 sourceDecimals;
+    uint8 destinationDecimals;
+}
+```
+
+ITS Hub can use the source and destination token decimals to determine the appropriate scaling factor to apply when intercepting this message. ITS edge contract can populate the source token decimals by reading the token metadata, however the destination token decimals will need to be provided by the deployer.
+
+Setting this incorrectly however can be lead to the ITS hub recording the deployment incorrectly, while the deployment tx on the destination ITS will fail due to mismatching decimals, which isn’t recoverable.
```

Comment:
why will it fail on the destination? Because the decimals would be greater than the chain supports? Can't we check this in the ITS hub?

Reply:
ITS hub doesn't know about the destination token's decimals. Solution 2 addresses this